homeassistant/components/jellyfin/remote.py

```diff
@@ -0,0 +1,113 @@
+    coordinator = jellyfin_data.coordinators["sessions"]
+
+    @callback
+    def handle_coordinator_update() -> None:
+        """Add remote per session."""
+        entities: list[RemoteEntity] = []
+        for session_id, session_data in coordinator.data.items():
+            if (
+                session_id not in coordinator.remote_session_ids
+                and session_data["SupportsRemoteControl"]
+            ):
+                entity: RemoteEntity = JellyfinRemote(
+                    coordinator, session_id, session_data
+                )
+                coordinator.remote_session_ids.add(session_id)
+                entities.append(entity)
+        async_add_entities(entities)
+
+    handle_coordinator_update()
+
+    entry.async_on_unload(coordinator.async_add_listener(handle_coordinator_update))
+
+
+class JellyfinRemote(JellyfinEntity, RemoteEntity):
+    """Defines a Jellyfin remote entity."""
+
+    def __init__(
+        self,
+        coordinator: JellyfinDataUpdateCoordinator,
+        session_id: str,
+        session_data: dict[str, Any],
+    ) -> None:
+        """Initialize the Jellyfin Remote entity."""
+        super().__init__(
+            coordinator,
+            RemoteEntityDescription(
+                key=session_id,
+            ),
+        )
+
+        self.session_id = session_id
+        self.session_data: dict[str, Any] = session_data
+        self.device_id: str = session_data["DeviceId"]
+        self.device_name: str = session_data["DeviceName"]
+        self.client_name: str = session_data["Client"]
+        self.app_version: str = session_data["ApplicationVersion"]
+
+        self.capabilities: dict[str, Any] = session_data["Capabilities"]
+
+        if self.capabilities.get("SupportsPersistentIdentifier", False):
+            self._attr_device_info = DeviceInfo(
+                identifiers={(DOMAIN, self.device_id)},
+                manufacturer="Jellyfin",
+                model=self.client_name,
+                name=self.device_name,
+                sw_version=self.app_version,
+                via_device=(DOMAIN, coordinator.server_id),
+            )
+            self._attr_name = None
+        else:
+            self._attr_device_info = None
+            self._attr_has_entity_name = False
+            self._attr_name = self.device_name
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the client is on."""
+        is_active: bool = self.session_data["IsActive"]
+        return is_active
+
+    def send_command(self, command: Iterable[str], **kwargs: Any) -> None:
+        """Send a command to the client."""
+        num_repeats = kwargs[ATTR_NUM_REPEATS]
+        delay = kwargs.get(ATTR_DELAY_SECS, DEFAULT_DELAY_SECS)
+
+        for _ in range(num_repeats):
+            for single_command in command:
+                self.coordinator.api_client.jellyfin.command(
+                    self.session_id, single_command
+                )
```

Comment:
I would like to be able to pass arguments for a command, but AFAICT `send_command` doesn't support this. Any ideas? For example, when sending `DisplayMessage` which accepts the `Header` and `Text` arguments.

Reply:
You can use action calls for these kind of calls

Reply:
Yeah, I wanted to use the provided remote command actions if at all possible but if not, likely need to create a "send_general_command" action.

Reply:
We don't like these as these don't provide context. (That's also why remote is a bit meh, but it works). So we would rather see a `display_message` action than a general catch all. With the specific services you can ask more contextual questions and parameters, increasing the user experience

Reply:
Sure, I was mostly just going with what I know (I use the Apple TV remote entity frequently). But if there's no way to send context to remote commands, then I agree that actions would be better. I'll add actions for the commands that take arguments.

Reply:
I'm not super familiar with how HA handles actions, is this the right approach (f3b325238ff731ec10ee915034b6f76a26d43be8)? If so, I'll add the rest of the commands with arguments as actions.

![image](https://github.com/user-attachments/assets/9705b9b5-63d1-4235-bbdc-22b7113a526d)